test/test_zeroinput.c

```diff
@@ -0,0 +1,96 @@
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <malloc.h>
+#include <string.h>
+#include <unistd.h>
+#include <zlib.h>
+
+#include "test.h"
+#include "test_utils.h"
+
+const char* flush2str[] = {
+	"Z_NO_FLUSH",
+	"Z_PARTIAL_FLUSH",
+	"Z_SYNC_FLUSH",
+	"Z_FULL_FLUSH",
+	"Z_FINISH",
+};
+
+int main()
+{
+	Byte *src, *compr;
+	unsigned int src_len = 2048;
+	unsigned int compr_len = src_len*3;
+	z_stream strm;
+	int failed = 0;
+
+	generate_random_data(src_len);
+	src = (Byte*)&ran_data[0];
+
+	compr = (Byte*)calloc((uInt)compr_len, 1);
+	if (compr == NULL || src == NULL) {
+		printf("*** alloc buffer failed\n");
+		return TEST_ERROR;
+	}
+
+	printf("-> empty data stream\n");
+	for(int flush = Z_NO_FLUSH; flush <= Z_FINISH; flush++) {
```

Comment:
This is cleaver but kind of confusing. A comment here could help.

Reply:
@lucmaga I added a comment to both loops that use this trick.